### ISSUE TITLE
Serialization turns bool array into uint8

### ISSUE THREAD

**Issue by rshin** [ray-project/ray#1121]
```
>>> import ray, numpy as np
>>> ray.init()
>>> np.array([True], dtype=bool)
array([ True], dtype=bool)
>>> ray.get(ray.put(np.array([True], dtype=bool)))
array([1], dtype=uint8)
```

**Comment by pcmoritz** [comment]
@rshin Thanks for reporting this! The offending line is this:https://github.com/apache/arrow/blob/9805ada853ae63580c59d765941f1e36a86e554c/cpp/src/arrow/python/numpy_convert.cc#L91

Will push a fix ASAP!

**Comment by pcmoritz** [comment]
see https://issues.apache.org/jira/browse/ARROW-1673

**Comment by robertnishihara** [comment]
Fixed by #1148, though it is handled inefficiently at the moment. That should be addressed soon in Arrow.
